bee-api/bee-rest-api/src/endpoints/routes/api/v2/milestone_by_milestone_index.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::net::IpAddr;
+
+use bee_message::payload::{
+    milestone::{MilestoneId, MilestoneIndex},
+    MilestonePayload,
+};
+use bee_runtime::resource::ResourceHandle;
+use bee_storage::access::Fetch;
+use bee_tangle::milestone_metadata::MilestoneMetadata;
+use log::error;
+use warp::{filters::BoxedFilter, reject, Filter, Rejection, Reply};
+
+use crate::{
+    endpoints::{
+        config::ROUTE_MILESTONE, filters::with_storage, path_params::milestone_index, permission::has_permission,
+        rejection::CustomRejection, storage::StorageBackend,
+    },
+    types::responses::MilestoneResponse,
+};
+
+fn path() -> impl Filter<Extract = (MilestoneIndex,), Error = Rejection> + Clone {
+    super::path()
+        .and(warp::path("milestones"))
+        .and(milestone_index())
+        .and(warp::path::end())
+}
+
+pub(crate) fn filter<B: StorageBackend>(
+    public_routes: Box<[String]>,
+    allowed_ips: Box<[IpAddr]>,
+    storage: ResourceHandle<B>,
+) -> BoxedFilter<(impl Reply,)> {
+    self::path()
+        .and(warp::get())
+        .and(has_permission(ROUTE_MILESTONE, public_routes, allowed_ips))
+        .and(with_storage(storage))
+        .and_then(|milestone_index, storage| async move { milestone_by_milestone_index(milestone_index, storage) })
+        .boxed()
+}
+
+pub(crate) fn milestone_by_milestone_index<B: StorageBackend>(
+    milestone_index: MilestoneIndex,
+    storage: ResourceHandle<B>,
+) -> Result<impl Reply, Rejection> {
+    let response = Fetch::<MilestoneIndex, MilestoneMetadata>::fetch(&*storage, &milestone_index).map_err(|e| {
```

Comment:
Were the tangle methods not enough ? It would be better to use them because when we reintroduce the tangle cache, using the storage directly would silently fail.

Reply:
fixed